### DISCUSSION TITLE
如何提升切换 DataGrid.ItemsSource 时的性能？

### DISCUSSION THREAD

**Discussion by CodingOctocat** [BYJRK/DotNet-Discussions#274] [WPF 开发]
简单来说，我开发了一个带数据库查询功能的程序，点击查询后，同时发起在线查询和本地缓存数据库查询，并将结果以 DataGrid 方式显示。

我的在线结果和本地结果共用一个视图，切换时仅仅是改变 `DataGrid.ItemsSource` 所绑定的属性（`OnlineVm.Results`、`LocalVm.Results`），但是我发现切换过程能感知到大约 <0.5s 的卡顿，有什么可用的方法提升性能？

**Comment by BYJRK** [comment]
你的数据量大概是什么规模？卡顿是加载看起来有延迟，还是界面也会短暂卡死？

**Reply by CodingOctocat** [reply]
查询得到的 Results 只有 10 条数据，切换时界面会卡死，DataGrid 有 5 列，ItemTemplate 总共有 9 个 TextBox 和 2 个 ListBox（也是 TextBox）（用的是 HandyControl 的带 Title 的 TextBox ），我怀疑是 DataGrid 渲染的问题。

**Reply by BYJRK** [reply]
这感觉有点不符合直觉哈。有没有一个实例代码，可以复现这个效果的？还是说随便一个例子，只要涉及到 ItemsSource 切换，就一定会卡顿？

**Reply by CodingOctocat** [reply]
基本可以确认是 DataGrid 渲染导致的卡顿，因为我试着展开其他布局，让 DataGrid 展示高度变小直至完全脱离视口，卡顿也逐步减少直至消失，似乎没有太好的解决方法。
